src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -12,16 +13,41 @@ namespace Microsoft.Extensions.Logging.Console
     [UnsupportedOSPlatform("browser")]
     internal class ConsoleLoggerProcessor : IDisposable
     {
-        private const int _maxQueuedMessages = 1024;
+        private static Meter _meter = new Meter("Microsoft-Extension-Logging-Console-Queue", "1.0.0");
+        private readonly Queue<LogMessageEntry> _messageQueue;
+        private int _messagesDropped;
+        private bool _isAddingCompleted;
+        private int _maxQueuedMessages = 1048576;
+        public int MaxQueueLength
+        {
+            get => _maxQueuedMessages;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentException(nameof(MaxQueueLength));
+                }
 
-        private readonly BlockingCollection<LogMessageEntry> _messageQueue = new BlockingCollection<LogMessageEntry>(_maxQueuedMessages);
+                lock (_messageQueue)
+                {
+                    _maxQueuedMessages = value;
+                    Monitor.PulseAll(_messageQueue);
```

Comment:
Is this meant to be reconfigurable at runtime? Should the `ConsoleLoggerProvider` be using `IOptionsMonitor` to call this setter when config changes?

Reply:
Yes this is supposed to be reconfigurable.
`ConsoleLoggerProvider` uses `IOptionsMonitor`:

https://github.com/dotnet/runtime/blob/24f5de4bb10976dfb391e9b6affe5481f55a2d00/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs#L41

In the reload callback we make sure to update the two properties in `ConsoleLoggerProcessor` accordingly.

TODO:

- [ ] Add test verifying it is reconfigurable.